sonar-html-plugin/src/main/java/org/sonar/plugins/html/lex/NormalElementTokenizer.java

```diff
@@ -33,16 +33,51 @@ public boolean consume(CodeReader codeReader, List<Node> nodeList) {
       return false;
     }
     int i = 1;
-    while (Character.isWhitespace(codeReader.charAt(i))) {
-      i++;
-    }
     if (codeReader.charAt(i) == '/' || codeReader.charAt(i) == '!') {
       i++;
     }
-    int elementNameIndex = i;
-    while (Character.isLetterOrDigit(codeReader.charAt(i))) {
-      i++;
+    return isValidTagNameStartChar(codeReader, i) && super.consume(codeReader, nodeList);
+  }
+
+  /**
+   * To avoid to wrongly consider a '<' character as being the start of a tag, the first character of the expected tag name is checked for validity.
+   * The ranges of valid starting characters for a tag name are defined here: https://www.w3.org/TR/REC-xml/#NT-NameStartChar.
+   * The method returns true in the following two cases:
+   *  - Current character code value is a valid single UTF-16 character codepoint in the expected range
+   *  - Current character and next character are a valid UTF-16 surrogate pair in the expected range
+   */
+  public static boolean isValidTagNameStartChar(CodeReader codeReader, int index) {
+    char nameStartChar = codeReader.charAt(index);
+    return isValidSingleCharCodeNameStartChar(nameStartChar) || isValidSurrogatePairNameStartChar(nameStartChar, codeReader.charAt(index + 1));
+  }
+
+  // Visible for testing
+  static boolean isValidSingleCharCodeNameStartChar(char currentChar) {
+    return currentChar == ':'
+      || (currentChar >= 'A' && currentChar <= 'Z')
+      || currentChar == '_'
+      || (currentChar >= 'a' && currentChar <= 'z')
+      || (currentChar >= '\u00C0' && currentChar <= '\u00D6')
+      || (currentChar >= '\u00D8' && currentChar <= '\u00F6')
+      || (currentChar >= '\u00F8' && currentChar <= '\u02FF')
+      || (currentChar >= '\u0370' && currentChar <= '\u037D')
+      || (currentChar >= '\u037F' && currentChar <= '\u1FFF')
+      || (currentChar >= '\u200C' && currentChar <= '\u200D')
+      || (currentChar >= '\u2070' && currentChar <= '\u218F')
+      || (currentChar >= '\u2C00' && currentChar <= '\u2FEF')
+      || (currentChar >= '\u3001' && currentChar <= '\uD7FF')
+      || (currentChar >= '\uF900' && currentChar <= '\uFDCF')
+      || (currentChar >= '\uFDF0' && currentChar <= '\uFFFD');
+  }
```

Comment:
_minor_ My Python 🐍 corrupted mind would prefer a construct like the one below, but it is not elegant as in Python ;). Anyway, I'm okay with your code.

```suggestion
  static final ImmutableRangeSet<Character> valid characters =  new ImmutableRangeSet.Builder<Character>()
    .add(Range.singleton(':'))
    .add(Range.closed('A', 'Z'))
    .add(Range.singleton('_'))
    .add(Range.closed('a', 'z'))
    .add(Range.closed('\u00C0','\u00D6'))
    .add(Range.closed('\u00D8','\u00F6'))
    .add(Range.closed('\u00F8','\u02FF'))
    .add(Range.closed('\u0370','\u037D'))
    .add(Range.closed('\u037F','\u1FFF'))
    .add(Range.closed('\u200C','\u200D'))
    .add(Range.closed('\u2070','\u218F'))
    .add(Range.closed('\u2C00','\u2FEF'))
    .add(Range.closed('\u3001','\uD7FF'))
    .add(Range.closed('\uF900','\uFDCF'))
    .add(Range.closed('\uFDF0','\uFFFD'))
    .build();
  static boolean isValidSingleCharCodeNameStartChar(char currentChar) {
    return valid characters.contains(currentChar);
  }
```

Reply:
I think your version is actually more readable.
The thing is that `ImmutableRangeSet` is from the `guava` library, and we try to avoid it as much as possible adding dependencies to keep the analyzers as lightweight as possible.
Another suggestion, to improve readability without adding a new dependency: how about having an additional `isInCharRange` helper method? That way the code would look like:
```
  static boolean isValidSingleCharCodeNameStartChar(char currentChar) {
    return currentChar == ':'
      || currentChar == '_'
      || isInCharRange(currentChar, 'A', 'Z')
      || isInCharRange(currentChar, 'a', 'z')
      || isInCharRange(currentChar, '\u00C0', '\u00D6')
      || isInCharRange(currentChar, '\u00D8', '\u00F6')
      || isInCharRange(currentChar, '\u00F8', '\u02FF')
      || isInCharRange(currentChar, '\u0370', '\u037D')
      || isInCharRange(currentChar, '\u037F', '\u1FFF')
      || isInCharRange(currentChar, '\u200C', '\u200D')
      || isInCharRange(currentChar, '\u2070', '\u218F')
      || isInCharRange(currentChar, '\u2C00', '\u2FEF')
      || isInCharRange(currentChar, '\u3001', '\uD7FF')
      || isInCharRange(currentChar, '\uF900', '\uFDCF')
      || isInCharRange(currentChar, '\uFDF0', '\uFFFD');
  }
```

Reply:
Understandable, alright. I prefer the first one at the end.

Reply:
Done: I've updated the code back to the original, more readable, version